src/hotrod/impl/operations/AddClientListenerOperation.cpp

```diff
@@ -78,7 +78,11 @@ char AddClientListenerOperation::executeOperation(transport::Transport& transpor
 			{
 				uint8_t status = codec20.readPartialEventHeader(transport, params);
 				if (!HotRodConstants::isSuccess(status))
-					break;
+				{
+				    auto errLen = transport.readVInt();
+				    auto errMsg = transport.readBytes(errLen);
+				    throw HotRodClientException(std::string(errMsg.begin(), errMsg.end()));
```

Comment:
@mgencur my patch is actually useless :)
The error code (0x50) returned by the server is documented both as error event and as a error response. The code actually precesses it as an error response and the exception is correctly propagated, no need for a fix.
On your c# test this exception is hidden by the one rised in the finally block. Please try with this chage:
```
            finally
            {
                if (cl.listenerId != null)
                {
                    cache.RemoveClientListener(cl);
                }
            }

```